test/integration/client-side-encryption/client_side_encryption.test.ts

```diff
@@ -0,0 +1,251 @@
+
+describe('Client-Side Encryption (Integration)', function () {
+  describe('CSOT', function () {
+    describe('Auto encryption', function () {
+      let setupClient;
+
+      beforeEach(async function () {
+        setupClient = this.configuration.newClient();
+        await setupClient
+          .db()
+          .admin()
+          .command({
+            configureFailPoint: 'failCommand',
+            mode: 'alwaysOn',
+            data: {
+              failCommands: ['aggregate'],
+              blockConnection: true,
+              blockTimeMS: 2000
+            }
+          } as FailPoint);
+      });
+
+      afterEach(async function () {
+        await setupClient
+          .db()
+          .admin()
+          .command({
+            configureFailPoint: 'failCommand',
+            mode: 'off'
+          } as FailPoint);
+        await setupClient.close();
+      });
+
+      context('when client is provided timeoutMS and command hangs', function () {
+        let encryptedClient;
+
+        beforeEach(async function () {
+          encryptedClient = this.configuration.newClient(
+            {},
+            {
+              autoEncryption: {
+                keyVaultNamespace: 'data.datakeys',
+                kmsProviders: {
+                  local: { key: Buffer.alloc(96) }
+                }
+              },
+              timeoutMS: 1000
+            }
+          );
+          await encryptedClient.connect();
+        });
+
+        afterEach(async function () {
+          await encryptedClient.close();
+        });
+
+        it(
+          'the command should fail due to a timeout error',
+          { requires: { mongodb: '>=4.2.0' } },
+          async function () {
+            const err = await encryptedClient
+              .db('test')
+              .collection('test')
+              .aggregate([])
+              .toArray()
+              .catch(e => e);
+            expect(err).to.be.instanceOf(MongoOperationTimeoutError);
+          }
+        );
+      });
+
+      context('when client is not provided timeoutMS and command hangs', function () {
+        let encryptedClient;
+        beforeEach(async function () {
+          encryptedClient = this.configuration.newClient(
+            {},
+            {
+              autoEncryption: {
+                keyVaultNamespace: 'admin.datakeys',
+                kmsProviders: {
+                  local: { key: Buffer.alloc(96) }
+                }
+              }
+            }
+          );
+        });
+
+        afterEach(async function () {
+          encryptedClient.close();
+        });
+
+        it('the command should not fail', { requires: { mongodb: '>=4.2.0' } }, async function () {
+          const err = await encryptedClient
+            .db('test')
+            .collection('test')
+            .aggregate([])
+            .toArray()
+            .catch(e => e);
+          expect(err).to.deep.equal([]);
```

Comment:
Usually, if we expect something not to fail, we don't catch the error because mocha considers a test to fail if an error is thrown:
```suggestion
    await encryptedClient
		.db('test')
		.collection('test')
		.aggregate([])
		.toArray()
```

Reply:
(this applies to other "should not fail" tests in this file too)

Reply:
Sounds good!